File: src/codes/Users_Manual/gain/holo_2.cpp
//~#include<autd3.hpp>
#include "autd3/gain/holo.hpp"

//~int main() {
const auto backend = std::make_shared<autd3::gain::holo::NalgebraBackend>();
auto g = autd3::gain::holo::GSPAT(backend).with_repeat(100);
//~return 0; }

File: src/codes/Users_Manual/gain/holo_2.py
~from pyautd3.gain.holo import GSPAT, NalgebraBackend
~from pyautd3.gain.holo import Pa
~import numpy as np
backend = NalgebraBackend()
~x1 = 0.0
~y1 = 0.0
~z1 = 0.0
~x2 = 0.0
~y2 = 0.0
~z2 = 0.0
~foci = [(np.array([x1, y1, z1]), 5e3 * Pa), (np.array([x2, y2, z2]), 5e3 * Pa)]
g = GSPAT(backend, foci).with_repeat(100)